dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -520,6 +525,20 @@ def get_workshop_for_new_general(enrollment_code, current_user)
       end
     end
 
+    # Pre surveys should be filled out by a teacher without attendance so
+    # look up nearest workshop that the current_user is enrolled in if
+    # enrollment_code is not given.
+    def get_workshop_for_pre_survey(enrollment_code, current_user)
+      if enrollment_code
+        Pd::Enrollment.find_by!(code: enrollment_code).workshop
+      else
+        Workshop.
+          where(course: [COURSE_CSD, COURSE_CSP]).
+          where.not(subject: SUBJECT_FIT).
+          nearest_enrolled_in_by(current_user)
```

Comment:
Is the `nearest_attended_or_enrolled_in_by` method poorly named? I would have thought that method does what you'd like (probably what you thought as well when you used it :).

Reply:
It's more stringent than named really. First it looks for the nearest workshop with attendance, then if it can't find one it looks for the nearest enrolled one. So a bit of a confusing name, yes

Reply:
I think such a change is probably out of scope for this bug fix, but maybe `nearest_attended_else_enrolled_in_by`? :)

Reply:
To be clear I'm talking about the old method, not your new one :) I guess I still don't understand, was the problem that the survey failed for teachers who have attended some other workshop in the past?

Reply:
Yeah the issue was if a teacher had attended another workshop previously it was trying to load a survey which didn't exist (ex. we don't have a foorm survey for academic year workshops). This was only an issue for the pre-survey because for everything else the teacher should have been marked attended for this workshop.

Reply:
👍 got it thanks!